File: javapatterns/src/main/java/org/iut/nantes/adapter/chimie/AdapterDB.java
package org.iut.nantes.adapter.chimie;

class AdapterDB extends OldDB
{
  private NewDB newDB_;

  public AdapterDB( NewDB newDB )
  {
    newDB_ = newDB;
  }

  public boolean exists( String name ) { return newDB_.exists( name ); }
  public Molecule oldMolecule( String name ) { return newDB_.newMolecule( name ); }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/chimie/AdapterDB.kt
package org.iut.nantes.adapter.chimie

internal class AdapterDB(private val newDB_: NewDB) : OldDB() {
    override fun exists(name: String?): Boolean {
        return newDB_.exists(name)
    }

    override fun oldMolecule(name: String?): Molecule? {
        return newDB_.newMolecule(name)
    }
}